packages/oauth/oauth-provider/src/request/request-manager.ts

```diff
@@ -457,53 +456,26 @@ export class RequestManager {
    * @note If this method throws an error, any token previously generated from
    * the same `code` **must** me revoked.
    */
-  public async findCode(
-    client: Client,
-    clientAuth: ClientAuth,
-    code: Code,
-  ): Promise<RequestDataAuthorized> {
-    const result = await this.store.findRequestByCode(code)
+  public async findCode(code: Code): Promise<RequestDataAuthorized> {
+    const result = await this.store.consumeRequestCode(code)
     if (!result) throw new InvalidGrantError('Invalid code')
 
     const { id, data } = result
-    try {
-      if (!isRequestDataAuthorized(data)) {
-        // Should never happen: maybe the store implementation is faulty ?
-        throw new Error('Unexpected request state')
-      }
 
-      if (data.clientId !== client.id) {
-        // Note: do not reveal the original client ID to the client using an invalid id
-        throw new InvalidGrantError(
-          `The code was not issued to client "${client.id}"`,
-        )
-      }
-
-      if (data.expiresAt < new Date()) {
-        throw new InvalidGrantError('This code has expired')
-      }
+    // A "code" can only be used once. This should already have been ensured
+    // by the store implementation, but let's be extra safe.
+    await this.store.deleteRequest(id)
```

Comment:
On the flip side, would this patch over an incorrect implementation and not surface the underlying issue?  I wonder if a warning log and/or assertion in dev mode might be more appropriate.